mycroft/tts/festival_tts.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import subprocess
+
+from .tts import TTS, TTSValidator
+
+
+class Festival(TTS):
+    def __init__(self, lang, config):
+        super(Festival, self).__init__(lang, config, FestivalValidator(self))
+
+    def execute(self, sentence, ident=None, listen=False):
+
+        toencoding = self.config.get('toencoding')
+
+        if toencoding != 'utf8':
+            cmd = "echo \"" + sentence + "\" | iconv -f utf8 -t " + toencoding + " | festival --tts --language " + self.lang
+        else:
+            cmd = "echo \"" + sentence + "\" | festival --tts --language " + self.lang
+
+        self.begin_audio()
+        subprocess.call(cmd, shell=True)
+        self.end_audio(listen)
+
+
+class FestivalValidator(TTSValidator):
+    def __init__(self, tts):
+        super(FestivalValidator, self).__init__(tts)
+
+    def validate_lang(self):
+        # TODO
```

Comment:
id also check thta lang is supported here, is there some command from festival to list languages or a hardcoded list?

Reply:
AFAIK, festival doesn't provide easy way to list of available languages from command line.

Any valid .scm file (private or custom voice, for instance) in voice festival directory can be set at --language switch.
For Debian based systems, it's `/usr/share/festival/languages/` or `/usr/share/festival/voice/`

So, validating "lang" parameter at this module can be really tricking.

Reply:
fair enough, it can be revisited in a follow up PR